src/main/java/com/example/cosmocats/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,63 @@
+package com.example.cosmocats.service.impl;
+
+import com.example.cosmocats.domain.Category;
+import com.example.cosmocats.repository.CategoryRepository;
+import com.example.cosmocats.repository.entity.CategoryEntity;
+import com.example.cosmocats.service.CategoryService;
+import com.example.cosmocats.service.exception.notFound.CategoryNotFoundException;
+import com.example.cosmocats.service.mapper.CategoryEntityMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.UUID;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryEntityMapper categoryEntityMapper;
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<Category> getAllCategories() {
+        return categoryRepository.findAll().stream()
+                .map(categoryEntityMapper::toCategory)
+                .toList();
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public Category getCategoryById(UUID id) {
+        CategoryEntity categoryEntity = categoryRepository.findById(id)
```

Comment:
problem here, you don't consider general exceptions that could potentially raise. Like db connectivity issue. So what you should do: try {} catch (Exception ex ) // that will hadle all exceptions coming from the category, than you should throw wrapper exception like PersistanceException. so later you can handle it accordingly and send to client 500 - DB connectivity lost or whatever. Also in case when you are throwing exceptions like here - you would add extra catch (CategoryNotFoundException ex) throw ex - so you will still throw the exception